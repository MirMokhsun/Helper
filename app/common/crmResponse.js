import { localization } from "./localization";

export const crmResponseMassage = (statusCode, errorFromServer, stor) => {
  const local = stor.getState().currentLanguage.language;
  const currentLang = localization[local];
  let statusResponse = '';
  if (statusCode === 401) {
    // Email or Password Incorrect
    statusResponse = currentLang.EmailOrPassWrong;
  }
  else if (statusCode === 500 && errorFromServer === 3) {
    // Invalid parametrs
    statusResponse = currentLang.InvalidParameters;
  }
  else if (statusCode === 500 && errorFromServer === 6) {
    // Email already exist
    statusResponse = currentLang.EmailIsExist;
  }
  else if (statusCode === 500 && errorFromServer === 10) {
    // Email not exist
    statusResponse = currentLang.EmailWrong;
  }
  else {
    // All another errors from server
    statusResponse = currentLang.AnotherProblem;
  }

  return statusResponse;
}