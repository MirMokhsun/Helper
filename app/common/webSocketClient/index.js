import WebSocketClient from './WebSocketClient';

let instance = null;
export class WebSocketSingleton {
    constructor(url) {
        if (!instance && url) {
            instance = this;
            this.webSocketClient = new WebSocketClient({ wsUrl: url });
        }
        return instance;
    }
}

// export const webSocketClient = new WebSocketSingleton().webSocketClient;
