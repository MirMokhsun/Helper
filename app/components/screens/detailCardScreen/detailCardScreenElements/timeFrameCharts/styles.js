import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        width: '90%',
        height: 32,
        backgroundColor: '#EBECEE',
        borderRadius: 16,
        padding: 4,
        marginBottom: 5,
        flexDirection: 'row',
        justifyContent: 'space-around',
        alignItems: 'center',
    },
    sliderInToogle: {
        width: '17%',
        height: 24,
        borderRadius: 12,
        backgroundColor: '#FFFFFF',
        zIndex: 1,
        position: 'absolute',
        top: 4,
        left: 4,
    },

});