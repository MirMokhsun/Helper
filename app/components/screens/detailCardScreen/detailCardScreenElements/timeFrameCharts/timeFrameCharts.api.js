import Utils from "../../../../../api/utils/utils";

const countButtons = 6;
const containerkWidth = 0.9 // 90%;
const paddingLeftAndRight = 8 // padding 4*2
export const buttonWidth = ((Utils.size.width * containerkWidth - paddingLeftAndRight) / countButtons);