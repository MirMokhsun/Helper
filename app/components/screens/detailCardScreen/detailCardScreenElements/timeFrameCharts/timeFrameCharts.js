import React, { PureComponent } from 'react';
import { View, Animated, Easing, I18nManager } from 'react-native';
import { styles } from './styles';
import TimeFrameButton from './timeFrameButton/timeFrameButton';
import { buttonWidth } from './timeFrameCharts.api';
import { RequestTecnivalAnalize } from '../../../../../api/setRequestTechnicalAnalize/setRequestTechnicalAnalize';
import store from '../../../../../redux/store/store';

const initialDataArray = [{ H: 1, L: 1, O: 1, C: 1, T: 0 }];
const zoom = { scaleX: 20, scaleY: 1, xValue: 300, yValue: 1, axisDependency: 'RIGHT' };
const zeroZoom = { scaleX: 0, scaleY: 0, xValue: 0, yValue: 0, axisDependency: 'RIGHT' };
const animationDuration = 100;

export class TimeFrameCharts extends PureComponent {
    constructor(props) {
        super(props);
        this.arrMultiple = I18nManager.isRTL ? [0, -1, -2, -3, -4, -5] : [0, 1, 2, 3, 4, 5];
        this.state = {
            leftDistance: new Animated.Value(buttonWidth * this.arrMultiple[3]),
            buttonFrame: {
                Minute: false,
                Minute5: false,
                Minute30: false,
                Hour: true,
                Day: false,
                Week: false,
            },
            currentDistance: 0
        };
        this.requestTechnical = new RequestTecnivalAnalize();
    };

    animation = (_toDistance, _fromDistane) => {
        const { leftDistance } = this.state;
        Animated.timing(
            leftDistance,
            {
                fromValue: _fromDistane,
                toValue: _toDistance,
                duration: animationDuration,
                easing: Easing.linear,
                useNativeDriver: true
            }
        ).start();
    };

    getDistance = (frame) => {
        let distance = null;
        switch (frame) {
            case 'Minute':
                distance = buttonWidth * this.arrMultiple[0];
                break;
            case 'Minute5':
                distance = buttonWidth * this.arrMultiple[1];
                break;
            case 'Minute30':
                distance = buttonWidth * this.arrMultiple[2];
                break;
            case 'Hour':
                distance = buttonWidth * this.arrMultiple[3];
                break;
            case 'Day':
                distance = buttonWidth * this.arrMultiple[4];
                break;
            case 'Week':
                distance = buttonWidth * this.arrMultiple[5];
                break;
            default:
                /* code */
                break;
        }
        return distance;
    };

    changeFrame = async (frame) => {
        try {
            const { cleanDatesArr } = this.props;
            const { currentDistance } = this.state;
            const distance = await this.getDistance(frame);
            await this.animation(distance, currentDistance);
            await this.setState({ currentDistance: distance });
            await this.setActiveButton(frame);
            await this.sendReqeust(frame);
            await cleanDatesArr();
        } catch (error) {
            console.log();
        }
    };

    setActiveButton = (frame) => {
        const buttonFrame = {
            Minute: false,
            Minute5: false,
            Minute30: false,
            Hour: false,
            Day: false,
            Week: false,
        };
        if (frame) {
            buttonFrame[frame] = true;
        };
        this.setState({ buttonFrame });
    };

    sendReqeust = async (frame) => {
        const { ID, setZoomCharts } = this.props;
        store.getState().candlesFunc.func(initialDataArray);
        setZoomCharts(zeroZoom);
        try {
            await this.requestTechnical.getCandelsFromServer(ID, frame);
            await this.requestTechnical.sendMassegeWebSoket(ID, frame);
            await setTimeout(() => {
                setZoomCharts(zoom);
            }, 0);
        } catch (error) {
        };
    };

    render() {
        const { local } = this.props;
        const { leftDistance, buttonFrame: { Minute, Minute5, Minute30, Hour, Day, Week } } = this.state;
        return (
            <View style={styles.container} >
                <Animated.View style={[styles.sliderInToogle, { transform: [{ translateX: leftDistance }] }]} ></Animated.View>
                <TimeFrameButton id={'Minute'} text={'1' + local['min']} isActive={Minute} setFrameFunction={this.changeFrame} frame={'Minute'} />
                <TimeFrameButton id={'Minute5'} text={'5' + local['min']} isActive={Minute5} setFrameFunction={this.changeFrame} frame={'Minute5'} />
                <TimeFrameButton id={'Minute30'} text={'30' + local['min']} isActive={Minute30} setFrameFunction={this.changeFrame} frame={'Minute30'} />
                <TimeFrameButton id={'Hour'} text={'1' + local['Hour']} isActive={Hour} setFrameFunction={this.changeFrame} frame={'Hour'} />
                <TimeFrameButton id={'Day'} text={'1' + local['Day']} isActive={Day} setFrameFunction={this.changeFrame} frame={'Day'} />
                <TimeFrameButton id={'Week'} text={'1' + local['Week']} isActive={Week} setFrameFunction={this.changeFrame} frame={'Week'} />
            </View>
        );
    };
}

export default TimeFrameCharts;