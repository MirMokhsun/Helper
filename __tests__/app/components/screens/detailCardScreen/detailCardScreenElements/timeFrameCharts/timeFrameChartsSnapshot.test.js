import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import { localization } from '../../../../../../../app/common/localization';
import { TimeFrameCharts } from '../../../../../../../app/components/screens/detailCardScreen/detailCardScreenElements/timeFrameCharts/timeFrameCharts';

describe('TimeFrameCharts', () => {
    test('Create snapshot TimeFrameCharts', () => {
        const renderer = new ShallowRenderer()
        let timeFrameCharts = renderer.render(<TimeFrameCharts local={localization.ru} />);
        expect(timeFrameCharts).toMatchSnapshot();
    });
    test.each([
        ['Minute', 0],
        ['Minute5', 111.16666666666667],
        ['Minute30', 222.33333333333334],
        ['Hour', 333.5],
        ['Day', 444.6666666666667],
        ['Week', 555.8333333333334],
    ])('test function getDistance with parametr %s', (frame, expected) => {
        let timeFrameCharts = renderer.create(<TimeFrameCharts local={localization.ru} />);
        let getDistance = timeFrameCharts.getInstance().getDistance;
        let result = getDistance(frame);
        expect(result).toEqual(expected);
    });
    test('test function setActiveButton', () => {
        let timeFrameCharts = renderer.create(<TimeFrameCharts local={localization.ru} />);
        let setActiveButton = timeFrameCharts.getInstance().setActiveButton;
        expect(timeFrameCharts.getInstance().state.buttonFrame.Minute30).toEqual(false);
        setActiveButton('Minute30');
        expect(timeFrameCharts.getInstance().state.buttonFrame.Minute30).toEqual(true);
    });
    test('test function changeFrame', () => {
        let timeFrameCharts = renderer.create(<TimeFrameCharts local={localization.ru} />);
        let changeFrame = timeFrameCharts.getInstance().changeFrame;
        changeFrame('Minute5').then(
            () => {
                expect(timeFrameCharts.getInstance().state.buttonFrame.Minute5).toEqual(true);
            }
        );
    });
});