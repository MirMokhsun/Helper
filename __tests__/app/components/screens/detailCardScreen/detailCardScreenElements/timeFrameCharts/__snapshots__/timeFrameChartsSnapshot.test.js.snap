// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeFrameCharts Create snapshot TimeFrameCharts 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EBECEE",
      "borderRadius": 16,
      "flexDirection": "row",
      "height": 32,
      "justifyContent": "space-around",
      "marginBottom": 5,
      "padding": 4,
      "width": "90%",
    }
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 12,
          "height": 24,
          "left": 4,
          "position": "absolute",
          "top": 4,
          "width": "17%",
          "zIndex": 1,
        },
        Object {
          "transform": Array [
            Object {
              "translateX": 333.5,
            },
          ],
        },
      ]
    }
  />
  <TimeFrameButton
    frame="Minute"
    id="Minute"
    isActive={false}
    setFrameFunction={[Function]}
    text="1мин"
  />
  <TimeFrameButton
    frame="Minute5"
    id="Minute5"
    isActive={false}
    setFrameFunction={[Function]}
    text="5мин"
  />
  <TimeFrameButton
    frame="Minute30"
    id="Minute30"
    isActive={false}
    setFrameFunction={[Function]}
    text="30мин"
  />
  <TimeFrameButton
    frame="Hour"
    id="Hour"
    isActive={true}
    setFrameFunction={[Function]}
    text="1час"
  />
  <TimeFrameButton
    frame="Day"
    id="Day"
    isActive={false}
    setFrameFunction={[Function]}
    text="1день"
  />
  <TimeFrameButton
    frame="Week"
    id="Week"
    isActive={false}
    setFrameFunction={[Function]}
    text="1нед"
  />
</View>
`;
